lib_eio_linux/eio_stubs.c

```diff
@@ -40,3 +45,37 @@ CAMLprim value caml_eio_getrandom(value v_ba, value v_off, value v_len) {
   if (ret == -1) uerror("getrandom", Nothing);
   CAMLreturn(Val_long(ret));
 }
+
+struct linux_dirent64 {
+    ino64_t        d_ino;    /* 64-bit inode number */
+    off64_t        d_off;    /* 64-bit offset to next structure */
+    unsigned short d_reclen; /* Size of this dirent */
+    unsigned char  d_type;   /* File type */
+    char           d_name[]; /* Filename (null-terminated) */
+};
+
+CAMLprim value caml_eio_getdents(value v_fd) {
+  CAMLparam1(v_fd);
+  CAMLlocal1(result);
+  char buf[DIRENT_BUF_SIZE];
+  struct linux_dirent64 *d;
```

Comment:
Can't we just use the version from `dirent.h`?
```suggestion
  struct dirent64 *d;
```

Reply:
Ah, I didn't see this when looking for linux examples using `getdents64`, thank you for pointing it out :))